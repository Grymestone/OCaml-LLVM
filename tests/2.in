double do_math(int a, double b) {
  int x = a * 5 + 3
  double y = b + 5.0
}

int main() {
    do_math(3, 8.0)
    0
}
