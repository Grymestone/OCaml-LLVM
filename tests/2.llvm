; ModuleID = 'tests/2.llvm'
source_filename = "awesome jit"

@buf = global [4 x i8] c"%d\0A\00"

declare i32 @printf(i8*, ...)

; Function Attrs: noinline nounwind optnone sspstrong uwtable
define double @do_math(i32 %a, double %b) #0 {
entry:
  %multmp = mul i32 %a, 8
  %addtmp = fadd double %b, 5.000000e+00
  ret double %addtmp
}

; Function Attrs: noinline nounwind optnone sspstrong uwtable
define i32 @main() #0 {
entry:
  %calltmp = call double @do_math(i32 3, double 8.000000e+00)
  %0 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @buf, i32 0, i32 0), i32 2)
  ret i32 0
}

attributes #0 = { noinline nounwind optnone sspstrong uwtable }
