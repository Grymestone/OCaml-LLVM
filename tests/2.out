FuncDec: int do_math
	args:VarDec: int a
VarDec: double b

	code: VarDecExpr: int x = id(a)  *  5  +  3 
VarDecExpr: double y = id(b)  +  5.000000 

Expr methodcall do_math 3  8.000000 

Declaring function do_math
Setting variable x
Operator *
Variable a
